Updating README file (#15)
hashicorp-demoapp/hashicups-client-go: README.md

# Hashicups Client (Go)

> A Go client library is written to provide transient connection between [Terraform-provider-hashicups](https://github.com/hashicorp/terraform-provider-hashicups) and [product api application](https://github.com/hashicorp-demoapp/product-api-go). This library itself does not required to be compiled, rather it is being utilized by Hashicups provider as a dependency. 

Reviewer: ```suggestion
A Go client package provides a transient connection between [terraform-provider-hashicups](https://github.com/hashicorp/terraform-provider-hashicups) and [product-api](https://github.com/hashicorp-demoapp/product-api-go). You do not need to compile this package, rather the HashiCups provider uses it as a dependency. 
```

> <div align="center">
>     <img src="./docs/terraform-provider-diagram.jpeg" alt="drawing" width="80%"/>
> </div>
> 
> <br />  

Reviewer: ```suggestion
![Image Caption](./docs/terraform-provider-diagram.jpeg)
```

> Using this module, Hashicups provider will be able to establish new client and send the HTTP(s) request over to product api application to perform CRUD operations. It also handles data mapping from user's inputs to `models.go`. The Hashicups URL defined as `http://localhost:19090` and this can be configured [here](https://github.com/hashicorp-demoapp/hashicups-client-go/blob/main/client.go#L11) in case product API port has been manually configured differently. This module also handles API calls to sign up, sign in and sign out for users authentication to use this application.

Reviewer: ```suggestion
Using this package, the HashiCups provider will be able to establish new client and make HTTP(s) request to `product-api` to perform CRUD operations. It also handles data mapping from user's inputs to `models.go`. The HashiCups URL defaults to `http://localhost:19090`. You can configure it [here](https://github.com/hashicorp-demoapp/hashicups-client-go/blob/main/client.go#L11) in case product API port has been manually configured differently. This module also handles API calls for user authentication.
```

# Hashicups Client (Go)

> A Go client package provides a transient connection between [terraform-provider-hashicups](https://github.com/hashicorp/terraform-provider-hashicups) and [product-api](https://github.com/hashicorp-demoapp/product-api-go). You do not need to compile this package, rather the HashiCups provider uses it as a dependency. 

Reviewer 2: ```suggestion
A Go client package that provides a transient connection between [terraform-provider-hashicups](https://github.com/hashicorp/terraform-provider-hashicups) and [product-api](https://github.com/hashicorp-demoapp/product-api-go). You do not need to compile this package; the HashiCups provider uses it as a dependency. 
```

![Hashicups diagram](./docs/terraform-provider-diagram.jpeg)

> Using this module, Hashicups provider will be able to establish new client and send the HTTP(s) request over to product api application to perform CRUD operations. It also handles data mapping from user's inputs to `models.go`. The Hashicups URL defined as `http://localhost:19090` and this can be configured [here](https://github.com/hashicorp-demoapp/hashicups-client-go/blob/main/client.go#L11) in case that port has been configured differently. This module also handles API calls to sign up, sign in and sign out for users authentication to use this application.

Reviewer 2: ```suggestion
Using this module, the Hashicups provider establishes a new client and sends HTTP(s) requests to the product api application to perform CRUD operations. It also handles data mapping from user's inputs to `models.go`. The Hashicups URL defaults to `http://localhost:19090` and you can configure it [here](https://github.com/hashicorp-demoapp/hashicups-client-go/blob/main/client.go#L11) in case that port is already in use. This module also handles API calls to sign up, sign in and sign out for users authentication to use this application.
```
